api/privateAppUserToken.ts

```diff
@@ -0,0 +1,55 @@
+import http from '../http';
+
+const LOCALDEVAUTH_API_PRIVATE_APP_USER_TOKEN_PATH =
+  'localdevauth/v1/private-app/user-token';
+
+export type PrivateAppUserTokenResponse = {
+  userId: number;
+  portalId: number;
+  appId: number;
+  scopeGroups: Array<string>;
+  userTokenKey: string;
+  expiresAt: string;
+  clientId: string;
+};
+
+export async function fetchPrivateAppUserToken(
+  accountId: number,
+  appId: number
+): Promise<PrivateAppUserTokenResponse> {
+  return http.get(accountId, {
+    url: `${LOCALDEVAUTH_API_PRIVATE_APP_USER_TOKEN_PATH}/${appId}`,
+  });
+}
+
+export async function createPrivateAppUserToken(
+  accountId: number,
+  appId: number,
+  scopeGroups: Array<string> = [],
+  expiresAt?: string
+): Promise<PrivateAppUserTokenResponse> {
+  return http.post(accountId, {
+    url: `${LOCALDEVAUTH_API_PRIVATE_APP_USER_TOKEN_PATH}/${appId}`,
+    data: {
+      ...(scopeGroups.length && { scopeGroups: scopeGroups }),
```

Comment:
Does this `createPrivateAppUserToken()` function support `scopeGroups` just for future use cases? From what I can tell nothing currently passes any scope groups to this. Could this be switched to an optional argument? Seems like it would be fine for it to just be undefined.

Somewhat related, another formatting option here that might be a little more straightforward is `scopeGroups: scopeGroups.length ? scopeGroups : undefined`.